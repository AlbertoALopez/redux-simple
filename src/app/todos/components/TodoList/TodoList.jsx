import React, { propTypes } from 'react';
import TodoItem from '../TodoItem/TodoItem.jsx';

const TodoList = ({ todos, handleTodoClick }) => {
    return (
        <ul>
            {todos.map((todo) => {
                return (
                    <TodoItem
                        key={todo.id}
                        {...todo}
                        onClick={() => handleTodoClick(todo.id)}
                    />
                );
            })}
        </ul>
    );
};

TodoList.propTypes = {
    todos: React.PropTypes.array.isRequired,
    handleTodoClick: React.PropTypes.func.isRequired,
};

export default TodoList;
