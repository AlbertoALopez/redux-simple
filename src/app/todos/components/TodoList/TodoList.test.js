import React from 'react';
import renderer from 'react-test-renderer';
import TodoList from './TodoList';
import { mount } from 'enzyme';


function onClick() {
    console.log('test');
}

const toDoItems = [
    {
        text: 'Test1',
        completed: false,
        id: 0,
    },
    {
        text: 'Test2',
        completed: false,
        id: 1,
    },
];

test('TodoList should render', () => {
    const component = renderer.create(
        <TodoList
            todos={toDoItems}
            handleTodoClick={onClick}
        />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('TodoList should render an unordered list with 2 list items', () => {
    const wrapper = mount(
        <TodoList
            todos={toDoItems}
            handleTodoClick={onClick}
        />
    );

    expect(wrapper.find('ul').length).toBe(1);
    expect(wrapper.find('li').length).toBe(2);
});
