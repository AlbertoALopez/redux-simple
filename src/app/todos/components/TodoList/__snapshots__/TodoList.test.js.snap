exports[`test TodoList should render 1`] = `
<ul>
  <li
    onClick={[Function]}
    style={
      Object {
        "textDecoration": "none",
      }
    }>
    <div
      className="todo-item-text">
      Test1
    </div>
  </li>
  <li
    onClick={[Function]}
    style={
      Object {
        "textDecoration": "none",
      }
    }>
    <div
      className="todo-item-text">
      Test2
    </div>
  </li>
</ul>
`;
